model/flow/identity.go

```diff
@@ -508,3 +562,59 @@ func (il IdentityList) EqualTo(other IdentityList) bool {
 	}
 	return true
 }
+
+// Exists takes a previously sorted Identity list and searches it for the target value
+// This code is optimized, so the coding style will be different
+// target:  value to search for
+// CAUTION:  The identity list MUST be sorted prior to calling this method
+func (il IdentityList) Exists(target *Identity) bool {
+	return il.IdentifierExists(target.NodeID)
+}
+
+// Exists takes a previously sorted Identity list and searches it for the target value
+// target:  value to search for
+// CAUTION:  The identity list MUST be sorted prior to calling this method
+func (il IdentityList) IdentifierExists(target Identifier) bool {
```

Comment:
The implementation here is non-trivial. Are we sure that `bytes.Compare` is more expensive? I feel that using `bytes.Compare` would result in a much more concise implementation; it that is possible (?)

Reply:
It's a high-use area.  If it weren't performance-sensitive, then bytes.Compare() would be ok.  I would vote for not using bytes.Compare() for that reason.

Reply:
👍  
Thanks [your reply](https://github.com/onflow/flow-go/pull/1513#discussion_r759749823) to my previous comment was really helpful.